native~/Runtime/src/UnityLifetime.h

```diff
@@ -16,7 +16,7 @@ class UnityLifetime {
     // In the Editor, we must use DestroyImmediate because Destroy won't
     // actually destroy the object.
     if (!DotNet::UnityEditor::EditorApplication::isPlaying()) {
-      DotNet::UnityEngine::Object::DestroyImmediate(o);
+      DotNet::UnityEngine::Object::DestroyImmediate(o, true);
```

Comment:
When I first implemented the cusom material described above, and seleted it as the "Opaque Material" on the tileset, Unity would complain about deleting assets whenever I refreshed the tileset or switched back to the default material. I thought it might have to do with the original material being an asset, even though the copies for the tiles presumably aren't? But I can't reproduce the problem at all anymore, so I'll revert this.